vllm/model_executor/layers/quantization/utils/w8a8_utils.py

```diff
@@ -144,8 +144,7 @@ def apply_fp8_linear(
             if type(output) is tuple and len(output) == 2:
                 output = output[0]
 
-            return torch.narrow(output, 0, 0,
-                                input_2d.shape[0]).view(*output_shape)
+            return output[0:input_2d.shape[0], ...].view(*output_shape)
```

Comment:
I'm not sure how to think about the efficiency difference between these operations - if you are saying the latter is fast should we change all usages of `torch.narrow` in this file?

Reply:
> I'm not sure how to think about the efficiency difference between these operations - if you are saying the latter is fast should we change all usages of `torch.narrow` in this file?

I'm not sure how the perf comparison looks like between slice and `torch.narrow`. But `torch.compile` is producing different IR for those two:

`torch.narrow`:
```
call_function  _scaled_mm_2           aten._scaled_mm.default        (getitem_10, arg11_1, arg10_1, arg12_1, None, None, torch.float16)    {}
call_function  slice_6                aten.slice.Tensor              (_scaled_mm_2, 0, 0, arg1_1)                                          {}
call_function  auto_functionalized_5  auto_functionalized            (<OpOverload(op='_C.fused_add_rms_norm', overload='default')>,)       {'input': slice_6, 'residual': getitem_4, 'weight': arg13_1, 'epsilon': 1e-05}
call_function  getitem_12             <built-in function getitem>    (auto_functionalized_5, 1)                                            {}
call_function  getitem_13             <built-in function getitem>    (auto_functionalized_5, 2)                                            {}
call_function  empty_4                aten.empty.memory_format       ([arg1_1, 4096],)                                                     {'dtype': torch.float8_e4m3fnuz, 'device': device(type='cuda', index=0), 'pin_memory': False}
call_function  slice_scatter_1        aten.slice_scatter.default     (_scaled_mm_2, getitem_12, 0, 0, arg1_1)                              {}
call_function  slice_9                aten.slice.Tensor              (slice_scatter_1, 0, 0, arg1_1)                                       {}
call_function  auto_functionalized_6  auto_functionalized            (<OpOverload(op='_C.static_scaled_fp8_quant', overload='default')>,)  {'result': empty_4, 'input': slice_9, 'scale': arg14_1}
call_function  getitem_15             <built-in function getitem>    (auto_functionalized_6, 1)                                            {}
call_function  _scaled_mm_3           aten._scaled_mm.default        (getitem_15, arg15_1, arg14_1, arg16_1, None, None, torch.float16)    {}
```
slice:
```
call_function  _scaled_mm_2           aten._scaled_mm.default        (getitem_10, arg11_1, arg10_1, arg12_1, None, None, torch.float16)    {}
call_function  auto_functionalized_5  auto_functionalized            (<OpOverload(op='_C.fused_add_rms_norm', overload='default')>,)       {'input': _scaled_mm_2, 'residual': getitem_4, 'weight': arg13_1, 'epsilon': 1e-05}
call_function  getitem_12             <built-in function getitem>    (auto_functionalized_5, 1)                                            {}
call_function  getitem_13             <built-in function getitem>    (auto_functionalized_5, 2)                                            {}
call_function  empty_4                aten.empty.memory_format       ([arg1_1, 4096],)                                                     {'dtype': torch.float8_e4m3fnuz, 'device': device(type='cuda', index=0), 'pin_memory': False}
call_function  auto_functionalized_6  auto_functionalized            (<OpOverload(op='_C.static_scaled_fp8_quant', overload='default')>,)  {'result': empty_4, 'input': getitem_12, 'scale': arg14_1}
call_function  getitem_15             <built-in function getitem>    (auto_functionalized_6, 1)                                            {}
call_function  _scaled_mm_3           aten._scaled_mm.default        (getitem_15, arg15_1, arg14_1, arg16_1, None, None, torch.float16)    {}
```

And the extra three ops from `torch.compile` makes the kernel fusion invalid.

Reply:
Interesting!

Reply:
Nvidia usually goes [here](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/quantization/utils/w8a8_utils.py#L117), so it won't have this issue.

Reply:
Is this a co-op reshape? Because it might be related to the `RedundantReshapesPass` and we could expand the cases it supports to eliminate the extra ops or produces